test/controllers/api/commits_controller_test.rb

```diff
@@ -9,7 +9,7 @@ class CommitsControllerTest < ActionController::TestCase
       end
 
       test "#index returns a list of commits" do
-        commit = @stack.commits.last
+        commit = @stack.commits.reachable.last
```

Comment:
It's a side effect. Commits in Shipit have a `detached: boolean` attribute. When someone force push a tracked branch, some commits we recorded are not on any branch anymore, and we mark them as such.

And since this only happen on force pushes, the last commit of a stack was incidentally never detached.

But Pull Request do reuse the commit model, they point to a `head` that is a detached commit. Because of this it is now possible for the last commit to be detached which broke that assumption, and several tests.